### DISCUSSION TITLE
stencil: Switching organizations

### DISCUSSION THREAD

**Discussion by jaredallard** [Homebrew/discussions#5587] [Writing Formulae/Casks] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.19
ORIGIN: https://github.com/Homebrew/brew
HEAD: 1f9bd2de89883ec9a59db3eab0f5475bce06d5bc
Last commit: 2 days ago
Core tap JSON: 04 Sep 15:21 UTC
Core cask tap JSON: 04 Sep 15:21 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: nvim
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 3.3.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/bin/ruby
CPU: dodeca-core 64-bit arm_blizzard_avalanche
Clang: 15.0.0 build 1500
Git: 2.46.0 => /opt/homebrew/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 14.6.1-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.4
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  upx
```

### Description of issue

Hi! [I introduced the `stencil` formula a long time ago at my past employer.](https://github.com/Homebrew/homebrew-core/pull/100644) When I switched places, I [started maintaining a hard fork](https://github.com/rgst-io/stencil) which generally most individuals (except my past employer) have started using.

I'm wondering what the best course of action is here, I've [been advertising a tap](https://github.com/rgst-io/homebrew-tap), but I'd like to remove the need for that eventually. Is this something I should open a PR to change, or is it best to leave it as-is?

**Comment by p-linnane** [comment, marked as answer]
We can't make a change like this unless the original upstream points to your fork as its official successor.

**Reply by jaredallard** [reply]
Makes sense to me! I'll get in touch with them and see. Thank you!

**Reply by carlocab** [reply]
TBF there is precedent for something similar when Michael Sweet left Apple and forked CUPS in the process.[^1][^2]

But in that case multiple distros had already made the switch, which made doing the same an easier decision for us to make.

[^1]: https://formulae.brew.sh/formula/cups#default
[^2]: https://github.com/Homebrew/homebrew-core/commit/8a470b51a714fa8e5c08614c30858770463072dd

**Reply by jaredallard** [reply]
@p-linnane Is the change they made sufficient for this? https://github.com/getoutreach/stencil#stencil 🙌

**Reply by p-linnane** [reply]
Yes, that's good enough for me.

**Reply by jaredallard** [reply]
Awesome, last question before opening up an PR here soon. Is there a recommended way of handling version changes? The fork I maintain started at `v0`, so it's technically going backwards :'( 

If it's a huge pain, I'm not opposed to doing a `v1` (it's probably about time anyways...)

**Reply by carlocab** [reply]
> If it's a huge pain, I'm not opposed to doing a `v1` (it's probably about time anyways...)

It's kind of a pain. In a pinch, you can use `version_scheme`. But note that that never goes away once added.

**Reply by p-linnane** [reply]
If the version is less than the current formula, it will cause issues. As @carlocab said, we can add `version_scheme` to the formula to reset it, but that will need to remain in the formula as long as it's around. We try not to do this wherever possible.

**Reply by jaredallard** [reply]
I ended up waiting until we were ready to release a stable version and jumped ahead to v2 to make it easier on everyone involved! Opened up the PR here: https://github.com/Homebrew/homebrew-core/pull/203360. Thanks for all the help and advice as always y'all ❤️
